src/errorMiddleware.js

```diff
@@ -0,0 +1,22 @@
+module.exports = (hooks = {}) => async (error, req, res, next) => {
+  if (hooks.beforeSend) {
+    try {
+      await hooks.beforeSend(req, res)
+    } catch (beforeSendError) {
+      return next(beforeSendError)
+    }
+  }
+  if (res.headersSent) {
+    return next(error)
+  }
+  res.status(500).send({
+    error: error.message,
+  })
+  if (hooks.afterSend) {
+    try {
+      await hooks.afterSend(req, res)
+    } catch (afterSendError) {
+      return next(afterSendError)
```

Comment:
A bit of the same concern as before, although less relevant as we have already sent the `json` response by the time we get here. 

I am curious to know the use case that we are considering for this second hook. Would only one hook fulfill all use-cases?